src/interpret.ml

```diff
@@ -150,21 +162,21 @@ module Make (P : Interpret_intf.P) :
         | Sub -> Choice.return @@ sub n1 n2
         | Mul -> Choice.return @@ mul n1 n2
         | Div s -> begin
-          try
-            match s with
-            | S ->
-              let> overflow = Bool.and_ (eq n1 min_int) @@ eq n2 ~-(const 1l) in
-              if overflow then Choice.trap Integer_overflow
-              else Choice.return @@ div n1 n2
-            | U -> Choice.return @@ unsigned_div n1 n2
-          with Division_by_zero -> Choice.trap Integer_divide_by_zero
```

Comment:
I'm surprised that we don't need to catch the exception anymore in the concrete case

Reply:
Select works in both cases anyway so there is now an explicit guard marking the exception raising path unreachable